File: 44/forum_37348.jl
# https://projecteuler.net/thread=44;page=5#37348

function PE044(c=1)
  i = D = 0

  while c > 0
    D += 3 * i + 1
    i += 1
    Pd = 0

    for d in range(1:i)
      Pd += 3d - 2
      j = div(D - Pd, 3d)
      r = rem(D - Pd, 3d)
      if r > 0 && (1 + sqrt(1 + 24 * (j * (3 * j - 1)))) % 6
        print(D)
        c -= 1
      end
    end
  end
end

println(PE044(100))

File: 44/forum_37348.py
import math

def PE044(c=1): # c : how many wants you ?
  i = D = 0
  while c:
    D+=3*i+1 ; i+=1 ; Pd = 0
    for d in range(1, i):
      Pd += 3*d-2
      j, r = divmod(D-Pd, 3*d)
      if (not r) and not (1+math.sqrt(1+24*(j*(3*j-1)+D)))%6:
        print(D)
        c-=1

print(PE044(10000))